converter/converting.py

```diff
@@ -260,49 +256,37 @@ def mtab2usi_conversion(idf_file_path):
     """
     process_name = "mtab2usi_conversion"
 
-    current_dir = os.path.dirname(idf_file_path)
-    idf_file_name = os.path.basename(idf_file_path)
-    sdrf_file_path = None
-
     # Create logger
+    current_dir, idf_file_name = os.path.split(idf_file_path)
     logger = create_logger(current_dir, process_name, idf_file_name)
 
-    # Figure out the name and location of sdrf files
-    with codecs.open(idf_file_path, 'rU', encoding='utf-8') as f:
-        # U flag makes it portable across in unix and windows (\n and \r\n are treated the same)
-        for line in f:
-            if re.search(SDRF_FILE_NAME_REGEX, line):
-                sdrf_file_name = line.split("\t")[1].strip()
-                if os.path.exists(current_dir + DATA_DIRECTORY):
-                    sdrf_file_path = os.path.join(current_dir, DATA_DIRECTORY, sdrf_file_name)
-                else:
-                    sdrf_file_path = os.path.join(current_dir, sdrf_file_name)
-
-    logger.debug("Found SDRF file: {}".format(sdrf_file_path))
+    sdrf_file_path = get_sdrf_path(idf_file_path, logger)
 
     sub = data_objects_from_magetab(idf_file_path, sdrf_file_path)
 
     datamodel2json_conversion(sub, current_dir, logger)
 
 
-def data_objects_from_magetab(idf_file_path, sdrf_file_path):
+def data_objects_from_magetab(idf_file_path, sdrf_file_path, submission_type="sequencing"):
     """
     Parse IDF/SDRF files and transform metadata to common datamodel
 
     :param idf_file_path: string, path to IDF file
     :param sdrf_file_path: string, path to SDRF file
+    :param submission_type: string, microarray, sequencing or singlecell
     :return: Submission class object
     """
 
     study_info, protocols = parse_idf(idf_file_path)
-    samples, extracts, le, assays, raw_data, processed_data, is_microarray = parse_sdrf(sdrf_file_path)
+    samples, extracts, le, assays, raw_data, processed_data = parse_sdrf(sdrf_file_path)
```

Comment:
This kind of behaviour would be better served by a more object oriented approach (you return an object that gives access to these elements). This is of course a longer term goal more than something to fix for this merge.

Reply:
I agree this isn't the best solution. I could create a basic MAGE-TAB model class which holds these lists. 
For now I am leaving the MAGE-TAB parsing part in the "quick and dirty" mode as the main focus is on the reverse function.